src/cdf.rs

```diff
@@ -4958,136 +4977,112 @@ pub(crate) fn rav1d_cdf_thread_update(
-            for k in 0..$n1d {
-                update_bit_2d!($n2d, $n3d, $($name)+[k]);
-            }
-        }
-    }
+    update_cdf_3d!(2, 2, 4, coef.eob_bin_16);
+    update_cdf_3d!(2, 2, 5, coef.eob_bin_32);
+    update_cdf_3d!(2, 2, 6, coef.eob_bin_64);
+    update_cdf_3d!(2, 2, 7, coef.eob_bin_128);
+    update_cdf_3d!(2, 2, 8, coef.eob_bin_256);
+    update_cdf_2d!(2, 9, coef.eob_bin_512);
+    update_cdf_2d!(2, 10, coef.eob_bin_1024);
+    update_cdf_4d!(N_TX_SIZES, 2, 4, 2, coef.eob_base_tok);
+    update_cdf_4d!(N_TX_SIZES, 2, 41 /*42*/, 3, coef.base_tok);
+    update_cdf_4d!(4, 2, 21, 3, coef.br_tok);
+    update_cdf_4d!(N_TX_SIZES, 2, 11 /*22*/, 1, coef.eob_hi_bit);
+    update_cdf_3d!(N_TX_SIZES, 13, 1, coef.skip);
+    update_cdf_3d!(2, 3, 1, coef.dc_sign);
 
-    update_bit_1d!(BlockSize::COUNT, m.use_filter_intra);
-    update_cdf_1d!(4, m.filter_intra.0);
     for k in 0..2 {
         update_cdf_2d!(
             N_INTRA_PRED_MODES,
             N_UV_INTRA_PRED_MODES - 1 - (k == 0) as usize,
             m.uv_mode[k]
         );
     }
-    update_cdf_2d!(8, 6, m.angle_delta);
-    for k in 0..N_TX_SIZES - 1 {
-        update_cdf_2d!(3, cmp::min(k + 1, 2), m.txsz[k]);
-    }
-    update_cdf_3d!(2, N_INTRA_PRED_MODES, 6, m.txtp_intra1);
-    update_cdf_3d!(3, N_INTRA_PRED_MODES, 4, m.txtp_intra2);
-    update_bit_1d!(3, m.skip);
     for k in 0..BlockLevel::COUNT {
         update_cdf_2d!(4, dav1d_partition_type_count[k] as usize, m.partition[k]);
     }
-    update_bit_2d!(N_TX_SIZES, 13, coef.skip);
-    update_cdf_3d!(2, 2, 4, coef.eob_bin_16);
-    update_cdf_3d!(2, 2, 5, coef.eob_bin_32);
-    update_cdf_3d!(2, 2, 6, coef.eob_bin_64);
-    update_cdf_3d!(2, 2, 7, coef.eob_bin_128);
-    update_cdf_3d!(2, 2, 8, coef.eob_bin_256);
-    update_cdf_2d!(2, 9, coef.eob_bin_512);
-    update_cdf_2d!(2, 10, coef.eob_bin_1024);
-    update_bit_3d!(N_TX_SIZES, 2, 11 /*22*/, coef.eob_hi_bit);
-    update_cdf_4d!(N_TX_SIZES, 2, 4, 2, coef.eob_base_tok);
-    update_cdf_4d!(N_TX_SIZES, 2, 41 /*42*/, 3, coef.base_tok);
-    update_bit_2d!(2, 3, coef.dc_sign);
-    update_cdf_4d!(4, 2, 21, 3, coef.br_tok);
-    update_cdf_2d!(3, (RAV1D_MAX_SEGMENTS - 1) as usize, m.seg_id);
-    update_cdf_1d!(7, m.cfl_sign.0);
     update_cdf_2d!(6, 15, m.cfl_alpha);
-    update_bit_0d!(m.restore_wiener);
-    update_bit_0d!(m.restore_sgrproj);
-    update_cdf_1d!(2, m.restore_switchable.0);
-    update_cdf_1d!(3, m.delta_q.0);
-    update_cdf_2d!(5, 3, m.delta_lf);
-    update_bit_2d!(7, 3, m.pal_y);
-    update_bit_1d!(2, m.pal_uv);
+    update_cdf_2d!(2, 15, m.txtp_inter1);
+    update_cdf_1d!(11, m.txtp_inter2.0);
+    update_cdf_3d!(2, N_INTRA_PRED_MODES, 6, m.txtp_intra1);
+    update_cdf_3d!(3, N_INTRA_PRED_MODES, 4, m.txtp_intra2);
+    update_cdf_1d!(7, m.cfl_sign.0);
+    update_cdf_2d!(8, 6, m.angle_delta);
+    update_cdf_1d!(4, m.filter_intra.0);
+    update_cdf_2d!(3, (RAV1D_MAX_SEGMENTS - 1) as usize, m.seg_id);
```

Comment:
```suggestion
    update_cdf_2d!(3, SegmentId::COUNT - 1, m.seg_id);
```

When rebased on `main`.